pkg/kernel/Dockerfile

```diff
@@ -1,33 +1,79 @@
 # This file must be kept as much in sync with pkg/new-kernel/Dockerfile as posisble
-FROM lfedge/eve-alpine:145f062a40639b6c65efa36bed1c5614b873be52 AS kernel-build-base
-
-ENV BUILD_PKGS \
-    argp-standalone automake bash bc binutils-dev bison build-base \
-    diffutils flex git gmp-dev gnupg installkernel kmod elfutils-dev    \
-    linux-headers libunwind-dev mpc1-dev mpfr-dev ncurses-dev findutils \
-    openssl-dev patch rsync sed squashfs-tools tar xz xz-dev zlib-dev openssl \
-    attr-dev autoconf file coreutils libtirpc-dev libtool util-linux-dev
+# use the same set of packages for simplicity
+ARG BUILD_PKGS_BASE="argp-standalone automake bash bc binutils-dev bison build-base \
+                     diffutils flex git gmp-dev gnupg installkernel kmod elfutils-dev    \
+                     linux-headers libunwind-dev mpc1-dev mpfr-dev ncurses-dev findutils \
+                     openssl-dev patch rsync sed squashfs-tools tar xz xz-dev zlib-dev openssl \
+                     attr-dev autoconf file coreutils libtirpc-dev libtool util-linux-dev"
+
+FROM lfedge/eve-alpine:1fc941603ddc153f26104865db242e644c16aa99 as kernel-build-base-native
+ARG BUILD_PKGS_BASE
+RUN BUILD_PKGS="${BUILD_PKGS_BASE}" eve-alpine-deploy.sh
+
+# hadolint ignore=DL3029
+FROM --platform=${BUILDPLATFORM} lfedge/eve-alpine:1fc941603ddc153f26104865db242e644c16aa99 as kernel-build-base-cross
+ARG BUILD_PKGS_BASE
+RUN BUILD_PKGS="${BUILD_PKGS_BASE}" eve-alpine-deploy.sh
+
+# hadolint ignore=DL3029
+FROM --platform=${BUILDPLATFORM} lfedge/eve-cross-compilers:706fa9f570a366dc8bfc354504ed9547188101a2 AS cross-compilers
+
+# will use several packages from target arch and copy them to sysroot
+FROM lfedge/eve-alpine:1fc941603ddc153f26104865db242e644c16aa99 AS cross-compile-libs
+ENV PKGS musl-dev libgcc
 RUN eve-alpine-deploy.sh
 
+FROM kernel-build-base-cross AS kernel-cross-target-arm64
+ENV TARGET_ARCH=aarch64
+
+# install cross-compile packages and libs for target
+# hadolint ignore=DL3006
+FROM kernel-cross-target-${TARGETARCH} AS kernel-cross-build-amd64
```

Comment:
don't you need to use --platform=${BUILDPLATFORM} here as well? Are you installing cross compiler + libs into one single container, right?

Reply:
Correct, I install both into the image which based on `kernel-cross-target-${TARGETARCH}`, which one in turns based on `kernel-build-base-cross` and the last one based on `FROM --platform=${BUILDPLATFORM} lfedge/eve-alpine:3e225b14d659519459aefbd77cf9eb25070df594`. So, all of them inherit `--platform=${BUILDPLATFORM}`.